### PR TITLE
Cherry-picked commit for new memory model

### PR DESCRIPTION
**Pull request by anvacaru**

### CODE REVIEW

File: resources/lemmas.md [hunk 1]
```diff
@@ -103,8 +103,23 @@ The following lemmas essentially capture the signature extraction mechanisms.
...
+    rule #isRegularWordStack(.WordStack) => true
+
+    //Rules for #padToWidth with regular symbolic arguments.
+    //Same as for concrete #padToWidth, when WordStack is of regular form "A:B ... :.WordStack"
+    //Not clear why KEVM rules for #padToWidth were marked [concrete]. If they were general, rules below would not be necessary.
+    rule #padToWidth(N, WS) => WS
+      requires notBool #sizeWordStack(WS) <Int N andBool #isRegularWordStack(WS) ==K true
+
+    rule #padToWidth(N, WS) => #padToWidth(N, 0 : WS)
+      requires         #sizeWordStack(WS) <Int N andBool #isRegularWordStack(WS) ==K true
```
Comment (daejunpark) [inline resources/lemmas.md]: i don't think you need all these.

File: .build/.k.rev [hunk 2]
```diff
@@ -1 +1,5 @@
+<<<<<<< HEAD
 be708fe8ea35de3b4d54d21780548fc12b2faf20
+=======
+964eee2490bdb177b37f7069be14dae5fd6f91bd
+>>>>>>> 0217104... fix ambiguous lemma
```
Comment (daejunpark) [inline .build/.k.rev]: this is not correct. you have to properly resolve the cherry-pick conflict.

Comment (anvacaru) [reply .build/.k.rev]: indeed, thank you

File: resources/lemmas.md (hunk 1 above)
Comment (anvacaru) [reply resources/lemmas.md]: i'll remove them and update the pull request accordingly

**Comment by daejunpark** [comment]
you can merge if tests pass

**Pull request closed without merging**